### ISSUE TITLE
PERF401 false positive when appended list items depend on previous items

### ISSUE THREAD

**Issue by andersk** [astral-sh/ruff#6210]
This incorrectly triggers [PERF401 manual-list-comprehension](https://beta.ruff.rs/docs/rules/manual-list-comprehension/) (#5298):

```python
fibonacci = [0, 1]
for i in range(20):
    fibonacci.append(sum(fibonacci[-2:]))
print(fibonacci)
```

This cannot be converted to a list comprehension because each appended item depends on the previous items. Similar to #5581, but without an `if` conditional.

**Comment by charliermarsh** [comment]
Thanks, will fix.
